cmd/tokens.go

```diff
@@ -0,0 +1,124 @@
+package cmd
+
+import (
+	"errors"
+)
+
+var errMissingToken = errors.New("missing token")
+
+func init() {
+	register(&tokensCmd{})
+}
+
+type tokensCmd struct {
+	Create  createTokensCmd  `command:"create" description:"Create a new access token"`
+	List    lsTokensCmd      `command:"ls" description:"List available access tokens"`
+	Compare compareTokensCmd `command:"compare" description:"Check if access token is valid"`
+	Remove  rmTokensCmd      `command:"rm" description:"Remove a specific access token"`
+}
+
+func (x *tokensCmd) Name() string {
+	return "tokens"
+}
+
+func (x *tokensCmd) Short() string {
+	return "Manage Cafe developer access tokens"
+}
+
+func (x *tokensCmd) Long() string {
+	return `
+Tokens allow other peers to register with a Cafe peer.
+Use this command to create, list, compare, and remove tokens required for access to this peer's Cafe.
+`
+}
+
+type createTokensCmd struct {
+	Client ClientOptions `group:"Client Options"`
+}
+
+func (x *createTokensCmd) Usage() string {
+	return `
+
+Generates an access token (32 random bytes) and saves bcrypt encrypted version for future lookup.
+The response contains a base58 encoded version of the random bytes token.
+`
+}
+
+func (x *createTokensCmd) Execute(args []string) error {
+	setApi(x.Client)
+	res, err := executeStringCmd(POST, "tokens", params{})
+	if err != nil {
+		return err
+	}
+	output(res)
+	return nil
+}
+
+type lsTokensCmd struct {
+	Client ClientOptions `group:"Client Options"`
+}
+
+func (x *lsTokensCmd) Usage() string {
+	return `
+
+List info about all stored cafe developer tokens.`
+}
+
+func (x *lsTokensCmd) Execute(args []string) error {
+	setApi(x.Client)
+	var list []string
+	res, err := executeJsonCmd(GET, "tokens", params{}, &list)
+	if err != nil {
+		return err
+	}
+	output(res)
+	return nil
+}
+
+type compareTokensCmd struct {
+	Client ClientOptions `group:"Client Options"`
+}
+
+func (x *compareTokensCmd) Usage() string {
+	return `
+
+Check validity of existing cafe developer access token.
```

Comment:
Maybe should remove 'developer' nomenclature throughout? Doesn't _really_ mean much, as non-devs might also want to explicitly register with a given Cafe?

Reply:
Yeh, I'm gonna do this.